### PR TITLE
Import and use {rlang} type checks

### PR DESCRIPTION
**Pull request by teunbrand**
This PR solves an issue discussed off-github.

Briefly, in an effort to standardise message signalling, we should use the type checks that can be imported from {rlang}.
This PR imports these and uses them whenever a straightforward check was appropriate.

### CODE REVIEW

File: tests/testthat/_snaps/label-number.md [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
     Condition
       Error in `scale_cut()`:
-      ! `scale_cut` must be a named numeric vector
+      ! `scale_cut` must have names
     Code
       number(1, scale_cut = "x")
     Condition
       Error in `scale_cut()`:
-      ! `scale_cut` must be a named numeric vector
+      ! `breaks` must be a numeric vector, not the string "x".
```
Comment (thomasp85) [inline tests/testthat/_snaps/label-number.md:12]: This seems wrong

File: tests/testthat/test-label-bytes.R [hunk 2]
```diff
@@ -20,7 +20,7 @@ test_that("compatible with scale argument", {
 })
 
 test_that("errors if unknown unit", {
-  expect_error(label_bytes("unit")(0), "valid unit")
+  expect_error(label_bytes("unit")(0), "must be one of")
```
Comment (thomasp85) [inline tests/testthat/test-label-bytes.R:23]: let's make this a snapshot test while we touch it

**Review by thomasp85 — CHANGES_REQUESTED** [review]

**Comment by teunbrand** [comment]
CI failure might be unrelated:

> ✖ Failed to download sessioninfo 1.2.2 (x86_64-pc-linux-gnu-ubuntu-22.04)

File: tests/testthat/_snaps/label-number.md (hunk 1 above)
Comment (teunbrand) [reply tests/testthat/_snaps/label-number.md]: Good catch!

File: tests/testthat/test-label-bytes.R (hunk 2 above)
Comment (teunbrand) [reply tests/testthat/test-label-bytes.R]: Yep, agreed

**Comment by thomasp85** [comment]
Thanks

**Pull request merged**